common/djangoapps/third_party_auth/samlproviderdata/tests/test_samlproviderdata.py

```diff
@@ -141,16 +142,42 @@ def test_patch_one_provider_data(self):
 
     def test_delete_one_provider_data(self):
         # DELETE auth/saml/v0/providerdata/ -d data
-        url = reverse('saml_provider_data-detail', kwargs={'pk': self.saml_provider_data.id})
-        data = {}
-        data['enterprise_customer_uuid'] = ENTERPRISE_ID
+        url_base = reverse('saml_provider_data-detail', kwargs={'pk': self.saml_provider_data.id})
+        query_kwargs = {'enterprise_customer_uuid': ENTERPRISE_ID}
+        url = '{}?{}'.format(url_base, urlencode(query_kwargs))
         orig_count = SAMLProviderData.objects.count()
 
-        response = self.client.delete(url, data)
+        response = self.client.delete(url)
 
         self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
         self.assertEqual(SAMLProviderData.objects.count(), orig_count - 1)
 
         # ensure only the sso_url was updated
         query_set_count = SAMLProviderData.objects.filter(pk=self.saml_provider_data.id).count()
         self.assertEqual(query_set_count, 0)
+
+    def test_get_one_provider_data_failure(self):
+        set_jwt_cookie(self.client, self.user, [(ENTERPRISE_ADMIN_ROLE, BAD_ENTERPRISE_ID)])
+        self.client.force_authenticate(user=self.user)
+        url_base = reverse('saml_provider_data-list')
+        query_kwargs = {'enterprise_customer_uuid': BAD_ENTERPRISE_ID}
+        url = '{}?{}'.format(url_base, urlencode(query_kwargs))
+
+        response = self.client.get(url, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_unauthenticated_request_is_forbidden(self):
+        self.client.logout()
+        urlbase = reverse('saml_provider_data-list')
+        query_kwargs = {'enterprise_customer_uuid': ENTERPRISE_ID}
+        url = '{}?{}'.format(urlbase, urlencode(query_kwargs))
+        set_jwt_cookie(self.client, self.user, [(ENTERPRISE_LEARNER_ROLE, ENTERPRISE_ID)])
+        response = self.client.get(url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        # manually running second case as DDT is having issues.
+        self.client.logout()
```

Comment:
Why the second logout here?

Reply:
I am not sure how cookies get "unset". I couldn't find a method to unset them, so I was hoping a logout/login would do as such. That's also what this test _was_ doing when it was parameterized, but because I've had to separate the 2 cases instead of parameterizing them due to Jenkins issues that is why there is a lot of duplication here.

Reply:
Would be good to confirm, but I thought that the APITestCase's self.client would be reset per test already. not true? if you remove logout() does it not work as expected? I guess it does not hurt to logout to be sure however...

I do the authenticate with cookies per test though, assuming that will be reset per test, 

```
 def setUp(self):
        set_jwt_cookie(self.client, self.user, [(ENTERPRISE_ADMIN_ROLE, ENTERPRISE_ID)])
        self.client.force_authenticate(user=self.user)
```

Reply:
Ah I see now that you have to logout because at this point in the test, the user is already logged in, so I do not see issues with this, if this works ..

Reply:
but yes ideally this would be a separate test case ... I would not block pr on that though unless you find a way out of it .. perhaps setting up a separate test, then logging that use in again in that test will do?

Reply:
I mean logging out and logging in again in the same test vs logging out and in again in a different test case would be functionally equivalent. Ultimately I'm overriding the cookies we're setting so I don't really want to use what is established in setUp regardless.

So if the thing is that you'd _prefer_ these be two separate tests, then I can certainly separate them out. The logic was exactly the same between the tests so it made the most sense to be 1 test with parameters, but alas.

Reply:
got it, no worries I think the test makes sense as is too, so no blockers on this one from me. if test gets larger we can refactor it